opm/autodiff/RateConverter.hpp

```diff
@@ -493,11 +493,13 @@ namespace Opm {
 
                     // sum p, rs, rv, and T.
                     double hydrocarbonPV = pv_cell*hydrocarbon;
-                    pv += hydrocarbonPV;
-                    p += fs.pressure(FluidSystem::oilPhaseIdx).value()*hydrocarbonPV;
-                    rs += fs.Rs().value()*hydrocarbonPV;
-                    rv += fs.Rv().value()*hydrocarbonPV;
-                    T += fs.temperature(FluidSystem::oilPhaseIdx).value()*hydrocarbonPV;
+		    if(hydrocarbonPV>0){
```

Comment:
Basically here, if there is no hydrocarbon, the quantities will not be averaged.  I guess the code is for the similar cause with https://github.com/OPM/opm-simulators/pull/1961 .